rust/chains/tw_pactus/src/transaction.rs

```diff
@@ -0,0 +1,410 @@
+use std::fmt::Debug;
+use std::str::FromStr;
+
+use tw_coin_entry::error::prelude::{SigningError, SigningErrorType, SigningResult};
+use tw_hash::blake2::blake2_b;
+use tw_keypair::ed25519::sha512::{PrivateKey, PublicKey};
+use tw_keypair::ed25519::Signature;
+use tw_keypair::traits::SigningKeyTrait;
+use tw_proto::Pactus;
+
+use crate::address::Address;
+use crate::amount::Amount;
+use crate::encoder::error::Error as EncoderError;
+use crate::encoder::{deserialize, Decodable, Encodable};
+
+const VERSION_LATEST: u8 = 1;
+const FLAG_NOT_SIGNED: u8 = 0x02;
+
+#[derive(Debug, Clone)]
+pub enum PayloadType {
+    Transfer = 1,
+    Bond = 2,
+    Sortition = 3,
+    Unbond = 4,
+    Withdraw = 5,
+}
+
+impl TryFrom<u8> for PayloadType {
+    type Error = EncoderError;
+
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        match value {
+            1 => Ok(PayloadType::Transfer),
+            2 => Ok(PayloadType::Bond),
+            3 => Ok(PayloadType::Sortition),
+            4 => Ok(PayloadType::Unbond),
+            5 => Ok(PayloadType::Withdraw),
+            _ => Err(EncoderError::ParseFailed("Invalid PayloadType value")),
+        }
+    }
+}
+
+impl Encodable for PayloadType {
+    fn encode(&self, w: &mut dyn std::io::Write) -> Result<(), EncoderError> {
+        (self.clone() as u8).encode(w)
+    }
+
+    fn encoded_size(&self) -> usize {
+        1
+    }
+}
+
+impl Decodable for PayloadType {
+    fn decode(r: &mut dyn std::io::Read) -> Result<Self, EncoderError> {
+        PayloadType::try_from(u8::decode(r)?)
+    }
+}
+
+pub trait Payload: Debug + Encodable {
+    fn signer(&self) -> &Address;
+    fn value(&self) -> Amount;
+    fn payload_type(&self) -> PayloadType;
+}
+
+#[derive(Debug)]
+pub struct TransferPayload {
```

Comment:
Could you please move `TransferPayload` to a separate module?
For example,
```
/transaction/
/transaction/mod.rs
/transaction/payload/
/transaction/payload/mod.rs
/transaction/payload/transfer.rs
```

Reply:
It is done. Thanks for the recommendation.